src/main/scala/thurloe/service/ApiDataModels.scala

```diff
@@ -8,7 +9,22 @@ object ApiDataModelsJsonProtocol extends DefaultJsonProtocol {
   implicit val userKeyValuePairsFormat = jsonFormat2(UserKeyValuePairs)
 }
 
+object ThurloeQuery {
+  private def findParameters(wantedParameters: Seq[String], paramList: Seq[(String, String)]): Map[String, Option[String]] = {
+    val params2map = paramList.toMap
+    wantedParameters map { p => (p, params2map.get(p)) } toMap
+  }
+
+  def apply(params: Seq[(String, String)]): ThurloeQuery = {
+    val params2map = params.toMap
+    ThurloeQuery(params2map.get("userId"), params2map.get("key"), params2map.get("value"))
```

Comment:
This also silently ignores bad query parameters, which turns typos in key names into null filters